### DISCUSSION TITLE
Ability to move API Routes to `/api` instead of `/pages/api`

### DISCUSSION THREAD

**Discussion by ajones55555** [vercel/next.js#42079] [Ideas]
### Describe the feature you'd like to request

With the new /app folder, it's a little weird to have API routes staying in /pages/api

I know that /pages isn't going away, but virtually everything in /pages can be migrated to /app now except for API routes... new projects or migrated projects using /app for all FE code make the /pages directory seem like a weird remnant.

I think separate /app and /api folders will also make separate of backend and frontend more clear.

### Describe the solution you'd like

- Support a new folder next to /app called /api which allows users to set up API routes in the exact same way as before, but without /pages
- A codemod to move /pages/api to /api and update imports

### Describe alternatives you've considered

- The only alternative I thought of would be leaving /api in /pages where it is now. Of course this isn't horrible, but semantically weird for new projects using /app and /pages/api

**Comment by davidlashlee** [comment]
I agree.

**Comment by BWBama85** [comment]
Agreed.

**Comment by Deja-vuuu** [comment]
Agreed. /pages/api  very strange

**Comment by aokigit** [comment]
Yes please. Always hated having the API folder inside pages.

**Comment by yeliex** [comment]
is `/apis` better than `/app/api` or `/pages/api`

**Comment by ajones55555** [comment]
@yeliex it would be `/api`, not `/apis` (for consistency with the current `/pages/api`)

I didn't propose `/app/api` because I thought that would get confusing since API routes won't have the same nesting and special file/folder structure as actual FE pages in `/app`

**Comment by roman-kirsanov** [comment]
Agree.

This is how my project structure would look like:
```
📂 Project/
    📂 dist/
    📂 src/
        📂 api/
        📂 app/
        📂 lib/
            📂 components/
            ...  
        📂 model/
        📂 public/
        📂 styles/
        📂 types/ 
    📂 task/
    📄 package.json
    📄 next.config.js
    📄 tsconfig.json
    ...
```

**Comment by JairoRaudaDev** [comment]
I agree

**Comment by Comi9** [comment]
Based on all the discussion threads I've went through, there doesn't seem to be any _official_ recommendation with regards to the placement of the new **api** folder, yet. However, if we take a guess that addresses the 'lib' of https://app-dir.vercel.app/ which mostly handles the data provisions for this demo, then we can assume that the **api** will live outside the **app** folder. Had anyone tried yet the behaviour of a NodeJS backend fetch method outside a server component (that needs to reside in **app**), for now?
